app/src/main/java/com/umc/edison/presentation/edison/BubbleInputViewModel.kt

```diff
@@ -371,179 +338,116 @@ class BubbleInputViewModel @Inject constructor(
-                (bubble.contentBlocks[0].content.parseHtml()
-                    .isNotEmpty() && bubble.contentBlocks[0].content != "<br>")
-                        || (bubble.contentBlocks[1].content.parseHtml()
-                    .isNotEmpty() && bubble.contentBlocks[1].content != "<br>")
-            }
-        }
-
+        val canSave = contentBlockManager.checkCanSave()
         _uiState.update { it.copy(canSave = canSave) }
     }
 
     private fun trimBlankBlock() {
-        val contentBlocks = _uiState.value.bubble.contentBlocks.toMutableList()
-        val updatedContentBlocks = mutableListOf<ContentBlockModel>()
-
-        contentBlocks.forEachIndexed { _, contentBlock ->
-            if (contentBlock.type == ContentType.TEXT) {
-                val contents = contentBlock.content.split("<br>")
-                val newContent = if (contents.size > 1) {
-                    contents.joinToString(separator = "") { it.trim() }
-                } else {
-                    contentBlock.content
-                }
-
-                if (newContent.parseHtml().trim().isNotEmpty()) {
-                    updatedContentBlocks.add(contentBlock)
-                }
-            } else {
-                updatedContentBlocks.add(contentBlock)
-            }
-        }
-
-        _uiState.update {
-            it.copy(
-                bubble = it.bubble.copy(contentBlocks = updatedContentBlocks)
-            )
-        }
+        contentBlockManager.trimBlankBlocks()
     }
 
     fun openGallery() {
-        if (!checkCanAddImage()) {
+        val currImageSize = imageHandler.getCurrentImageCount(_uiState.value.bubble.contentBlocks)
+        if (!imageHandler.checkCanAddImage(currImageSize)) {
+            showToast(MAX_TOTAL_IMAGES_LIMIT_MESSAGE)
             return
         }
 
         _uiState.update { it.copy(isGalleryOpen = true) }
     }
 
     fun saveCameraImage(uri: Uri) {
-        _uiState.update {
-            it.copy(cameraImagePath = uri)
-        }
+        _uiState.update { it.copy(cameraImagePath = uri) }
     }
 
     fun saveCameraImage(context: Context) {
-        val savedUri = saveImageToInternalStorage(context, _uiState.value.cameraImagePath!!)
+        val savedUri = imageHandler.saveCameraImage(context, _uiState.value.cameraImagePath!!)
         _uiState.update {
             it.copy(
                 cameraImagePath = null,
                 isCameraOpen = false,
             )
         }
         addImagesToContentBlocks(
-            uris = listOf(savedUri)
+            imageUris = listOf(savedUri)
         )
     }
 
     fun updateCameraOpen(isOpen: Boolean) {
-        if (isOpen && !checkCanAddImage()) {
-            return
+        if (isOpen) {
+            val currImageSize = imageHandler.getCurrentImageCount(_uiState.value.bubble.contentBlocks)
+            if (!imageHandler.checkCanAddImage(currImageSize)) {
+                showToast(MAX_TOTAL_IMAGES_LIMIT_MESSAGE)
+            }
         }
```

Comment:
요렇게 해도 좋을 거 같아요! if문 depth가 너무 깊어지는 거 같아서..
```suggestion
        val currImageSize = imageHandler.getCurrentImageCount(_uiState.value.bubble.contentBlocks)
        if (isOpen && !imageHandler.checkCanAddImage(currImageSize)) {
            showToast(MAX_TOTAL_IMAGES_LIMIT_MESSAGE)
        }
```